esp-hal-common/src/gpio.rs

```diff
@@ -650,6 +630,16 @@ where
     PINTYPE: PinType,
     SIG: GpioSignal,
 {
+    pub fn new() -> Self {
```

Comment:
Remove this function, and use `steal()` instead wherever it is needed.

Reply:
Ahh thanks will do! That was just copied from when I did this work previously, which I think predated `steal` 😁

Reply:
`steal()` didn't exist so I've changed this to `pub(crate)` instead.